force-app/main/default/classes/SOQL.cls

```diff
@@ -313,19 +349,21 @@ public with sharing class SOQL {
      * @param List of cloned Opportunity Ids
      */
     public static void cloneLineItems(Id origOppId, List<Id> cloneOppIds) {
-        // Query the original Opportunity Line Items
-
-        // Create a list to hold the new line items
+        List<OpportunityLineItem> foundOppLineItems = [SELECT Id, OpportunityId, Quantity, PricebookEntryId, UnitPrice FROM OpportunityLineItem WHERE OpportunityId = :origOppId]; // Query the original Opportunity Line Items
+        List<OpportunityLineItem> newOppLineItems = new List<OpportunityLineItem>(); // Create a list to hold the new line items
 
-        // Generate a random number between 1 and 3
+        for(Id i : cloneOppIds) { // Loop through each cloned Opportunity Id
+            for(OpportunityLineItem olt : foundOppLineItems){ // Loop through each original line items
+                Integer randomNum = (Integer)(Math.random()*3+1); // Find a random number for each of the new Opp Line Item using formula: (Integer)(Math.random()*(max-min+1)+1)
+                OpportunityLineItem newLineItem = olt.clone(false, true, false, false); // Clone the original line item
```

Comment:
I saw your feedback on this. You dont have to use the .clone() method to make this work, but it is the best way to do this, so good job finding it.